frontend/src/metabase/admin/permissions/components/PermissionsPageLayout/PermissionsPageLayout.tsx

```diff
@@ -23,38 +21,40 @@ import {
   PermissionPageSidebar,
   CloseSidebarButton,
   ToolbarButtonsContainer,
-} from "./PermissionsPageLayout.styled";
-import { PermissionsEditBar } from "./PermissionsEditBar";
+} from "metabase/admin/permissions/components/PermissionsPageLayout/PermissionsPageLayout.styled";
+import { useLeaveConfirmation } from "../../hooks/use-leave-confirmation";
+import { clearSaveError as clearPermissionsSaveError } from "../../permissions";
+import { ToolbarButton } from "../ToolbarButton";
+import { PermissionsTabs } from "./PermissionsTabs";
 
-const mapDispatchToProps = {
-  navigateToTab: tab => push(`/admin/permissions/${tab}`),
-  navigateToLocation: location => push(location.pathname, location.state),
-  clearSaveError,
-};
+import { PermissionsEditBar } from "./PermissionsEditBar";
 
-const mapStateToProps = (state, _props) => {
-  return {
-    saveError: state.admin.permissions.saveError,
-  };
+type PermissionsPageTab = "data" | "collections";
+type PermissionsPageLayoutProps = {
+  children: React.ReactNode;
+  tab: PermissionsPageTab;
+  confirmBar?: React.ReactNode;
+  diff?: PermissionsGraph;
+  isDirty: boolean;
+  onSave: () => void;
+  onLoad: () => void;
+  saveError?: string;
+  clearSaveError: () => void;
+  navigateToLocation: (location: string) => void;
+  router: typeof Router;
+  route: typeof Route;
+  navigateToTab: (tab: string) => void;
+  helpContent?: React.ReactNode;
+  toolbarRightContent?: React.ReactNode;
 };
 
-const propTypes = {
-  children: PropTypes.node.isRequired,
-  tab: PropTypes.oneOf(["data", "collections"]).isRequired,
-  confirmBar: PropTypes.node,
-  diff: PropTypes.object,
-  isDirty: PropTypes.bool,
-  onSave: PropTypes.func.isRequired,
-  onLoad: PropTypes.func.isRequired,
-  saveError: PropTypes.string,
-  clearSaveError: PropTypes.func.isRequired,
-  navigateToLocation: PropTypes.func.isRequired,
-  router: PropTypes.object,
-  route: PropTypes.object,
-  navigateToTab: PropTypes.func.isRequired,
-  helpContent: PropTypes.node,
-  toolbarRightContent: PropTypes.node,
-};
+const CloseSidebarButtonWithDefault = ({
```

Comment:
I added this because I kept running into type errors on the test with the `CloseSidebarButton` component - it seems like TS doesn't check `defaultProps` in styled components. From what I've seen on SO, there only seems to be workarounds like this until the linter is able to recognize `defaultProps` that are dynamically added to components